integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -19,6 +20,215 @@
+                        arguments=tool_use.get("input", {}),
+                    )
+                    tool_calls.append(tool_call)
+
+            # Create a single ChatMessage with combined text and tool calls
+            replies.append(ChatMessage.from_assistant(" ".join(text_content), tool_calls=tool_calls, meta=base_meta))
+
+    return replies
+
+
+def _parse_bedrock_streaming_chunks(
+    response_stream: EventStream,
+    streaming_callback: Callable[[StreamingChunk], None],
+    model: str,
+) -> List[ChatMessage]:
+    """
+    Parse a streaming response from Bedrock.
+
+    :param response_stream: EventStream from Bedrock API
+    :param streaming_callback: Callback for streaming chunks
+    :param model: The model ID used for generation
+    :return: List of ChatMessage objects
+    """
+    replies = []
+    current_content = ""
+    current_tool_call: Optional[Dict[str, Any]] = None
+    base_meta = {
+        "model": model,
+        "index": 0,
+    }
+
+    for event in response_stream:
+        if "contentBlockStart" in event:
+            # Reset accumulators for new message
+            current_content = ""
+            current_tool_call = None
+            block_start = event["contentBlockStart"]
+            if "start" in block_start and "toolUse" in block_start["start"]:
+                tool_start = block_start["start"]["toolUse"]
+                current_tool_call = {
+                    "id": tool_start["toolUseId"],
+                    "name": tool_start["name"],
+                    "arguments": "",  # Will accumulate deltas as string
+                }
+
+        elif "contentBlockDelta" in event:
+            delta = event["contentBlockDelta"]["delta"]
+            if "text" in delta:
+                delta_text = delta["text"]
+                current_content += delta_text
+                streaming_chunk = StreamingChunk(content=delta_text, meta=None)
+                streaming_callback(streaming_chunk)
+            elif "toolUse" in delta and current_tool_call:
+                # Accumulate tool use input deltas
+                current_tool_call["arguments"] += delta["toolUse"].get("input", "")
+
+        elif "contentBlockStop" in event:
+            if current_tool_call:
+                # Parse accumulated input if it's a JSON string
+                try:
+                    input_json = json.loads(current_tool_call["arguments"])
+                    current_tool_call["arguments"] = input_json
+                except json.JSONDecodeError:
+                    # Keep as string if not valid JSON
+                    pass
+
+                tool_call = ToolCall(
+                    id=current_tool_call["id"],
+                    tool_name=current_tool_call["name"],
+                    arguments=current_tool_call["arguments"],
+                )
+                replies.append(ChatMessage.from_assistant("", tool_calls=[tool_call], meta=base_meta.copy()))
```

Comment:
same comment as above: `ChatMessage.from_assistant(text="", ...)` is handled properly by Bedrock?

Reply:
Yes, double checked this - no issues, see above for details